keras_cv/models/regnet.py

```diff
@@ -0,0 +1,1529 @@
+        "block_type": "Y",
+    },
+}
+
+BASE_DOCSTRING = """Instantiates the {name} architecture.
+
+  Reference:
+    - [Designing Network Design Spaces](https://arxiv.org/abs/2003.13678)
+    (CVPR 2020)
+
+  For image classification use cases, see
+  [this page for detailed examples](
+  https://keras.io/api/applications/#usage-examples-for-image-classification-models).
+
+  For transfer learning use cases, make sure to read the
+  [guide to transfer learning & fine-tuning](
+    https://keras.io/guides/transfer_learning/).
+
+
+  The naming of models is as follows: `RegNet<block_type><flops>` where
+  `block_type` is one of `(X, Y)` and `flops` signifies hundred million
+  floating point operations. For example RegNetY064 corresponds to RegNet with
+  Y block and 6.4 giga flops (64 hundred million flops).
+
+  Args:
+    include_rescaling: whether or not to Rescale the inputs.If set to True,
+        inputs will be passed through a `Rescaling(1/255.0)` layer.
+    include_top: Whether to include the fully-connected
+        layer at the top of the network.
+    num_classes: Optional number of classes to classify images
+        into, only to be specified if `include_top` is True, and
+        if no `weights` argument is specified.
+    weights: One of `None` (random initialization), or the path to the weights
+          file to be loaded. Defaults to `None`.
+    input_shape: Optional shape tuple, defaults to (None, None, 3).
+        It should have exactly 3 inputs channels.
+    pooling: Optional pooling mode for feature extraction
+        when `include_top` is `False`. Defaults to None.
+        - `None` means that the output of the model will be
+            the 4D tensor output of the
+            last convolutional layer.
+        - `avg` means that global average pooling
+            will be applied to the output of the
+            last convolutional layer, and thus
+            the output of the model will be a 2D tensor.
+        - `max` means that global max pooling will
+            be applied.
+    classifier_activation: A `str` or callable. The activation function to use
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+        Defaults to `"softmax"`.
+        When loading pretrained weights, `classifier_activation` can only
+        be `None` or `"softmax"`.
+
+  Returns:
+    A `keras.Model` instance.
+"""
+
+
+def Stem(name=None):
+    """Implementation of RegNet stem.
+
+    (Common to all model variants)
+    Args:
+      name: name prefix
+
+    Returns:
+      Output tensor of the Stem
+    """
+    if name is None:
+        name = "stem" + str(backend.get_uid("stem"))
+
+    def apply(x):
+        x = layers.Conv2D(
+            32,
+            (3, 3),
+            strides=2,
+            use_bias=False,
+            padding="same",
+            kernel_initializer="he_normal",
+            name=name + "_stem_conv",
```

Comment:
what does a stem do?  classification?  Outputs before classification?

Reply:
`Stem` is a set of layers common to all RegNet architectures, applied before the `Stages`. See attached image.

![image](https://user-images.githubusercontent.com/64411306/180858857-f2afd4dd-1bfa-4be0-a4a5-84b5656e57bc.png)